File: job-hunt-frontend/job-hunt-ui/src/services/LocationService.js
import axios from "axios";
import { getToken } from "./AuthService";

const REST_API_LOCATION_BASE_URL = 'http://localhost:8080/api/locations';

// Add a request interceptor
axios.interceptors.request.use(function (config) {
    config.headers['Authorization'] = getToken();
    return config;
  }, function (error) {
    return Promise.reject(error);
  });


export const getAllLocations = () => axios.get(REST_API_LOCATION_BASE_URL); // GET ALL

export const addLocation = (location) => axios.post(REST_API_LOCATION_BASE_URL, location) // ADD

export const getLocation = (id) => axios.get(REST_API_LOCATION_BASE_URL + '/' + id); // GET

export const updateLocation = (id, location) => axios.put(REST_API_LOCATION_BASE_URL + '/'+id, location); // UPDATE

export const deleteLocation = (id) => axios.delete(REST_API_LOCATION_BASE_URL + '/' +id); // DELETE

File: job-hunt/src/main/java/ykpsph/jobhunt/service/LocationService.java
package ykpsph.jobhunt.service;

import ykpsph.jobhunt.dto.LocationDTO;

import java.util.List;

public interface LocationService {

    LocationDTO addLocation(LocationDTO locationDTO); // ADD

    List<LocationDTO> getAllLocations(); // GET ALL

    LocationDTO getLocation(Long id); // GET


    LocationDTO updateLocation(LocationDTO locationDTO, Long id); // UPDATE

    void deleteLocation(Long id); // DELETE
}
